src/lib/components/SquareButtonWithLoadingState.svelte

```diff
@@ -0,0 +1,27 @@
+<script lang="ts">
+	interface Props {
+		cssClass?: string;
+		onClick: () => void | Promise<void>;
+		icon: string;
+		disabled?: boolean;
+		duotone?: boolean;
+	}
+
+	let { cssClass, onClick, icon, disabled, duotone = true }: Props = $props();
+
+	let loading = $state(false);
+
+	const fullOnClick = async () => {
+		loading = true;
+		await onClick();
+		loading = false;
+	};
+</script>
+
+<button class="btn btn-square btn-sm {cssClass}" onclick={fullOnClick} {disabled}>
+	{#if loading}
+		<i class="fa-{duotone ? 'duotone' : 'solid'} fa-spinner fa-spin"></i>
+	{:else}
+		<i class="fa-{duotone ? 'duotone' : 'solid'} fa-{icon.replace('fa-', '')}"></i>
```

Comment:
Just curious: Why the replace call? :D

Reply:
I always confuse where I implemented the fontAwesome icon to be "fa-icon" or just "icon" – in other words, does the css class need the "fa-" prefix or is it already present. When you copy from the fontawesome website, it does not provide you with the prefix, but being so used to it I often include it anyways which breaks the icon.

In order to fix this while not trusting myself to actually remember how to do it, I thought having the prefix inside the class and omiting the "fa-" if present in the icon-string, I would solve this problem for good. So now: wherever I implement Icons in components, where the icons are not fixed but rather a prop, I implement it this way :D